Modules/posixmodule.c

```diff
@@ -5490,12 +5490,87 @@ os__path_normpath_impl(PyObject *module, PyObject *path)
         return NULL;
     }
     Py_ssize_t norm_len;
-    wchar_t *norm_path = _Py_normpath_and_size(buffer, len, &norm_len);
+    wchar_t *norm_path = _Py_normpath_and_size(buffer, len, 0, &norm_len);
     PyObject *result = PyUnicode_FromWideChar(norm_path, norm_len);
     PyMem_Free(buffer);
     return result;
 }
 
+#ifndef MS_WINDOWS
+
+/*[clinic input]
+os._path_abspath
+
+    path: object
+    /
+
+Make path absolute.
+[clinic start generated code]*/
+
+static PyObject *
+os__path_abspath(PyObject *module, PyObject *path)
+/*[clinic end generated code: output=6a6bb40f1ebe86f2 input=6f97dcc93c4bd71f]*/
+{
+    if (!PyUnicode_Check(path)) {
+        PyErr_Format(PyExc_TypeError, "expected 'str', not %T",
+                     Py_TYPE(path));
+        return NULL;
+    }
+
+    Py_ssize_t rel_path_len;
+    wchar_t *rel_path = PyUnicode_AsWideCharString(path, &rel_path_len);
+    if (!rel_path) {
+        return NULL;
+    }
+
+    Py_ssize_t cwd_len;
+    wchar_t *abs_path;
+    Py_ssize_t abs_path_len;
+    if (_Py_isabs(rel_path)) {
+        cwd_len = 0;
+        abs_path = rel_path;
+        abs_path_len = rel_path_len;
+    } else {
+        wchar_t cwd[MAXPATHLEN + 1];
+        cwd[Py_ARRAY_LENGTH(cwd) - 1] = 0;
+        if (!_Py_wgetcwd(cwd, Py_ARRAY_LENGTH(cwd) - 1)) {
+            /* unable to get the current directory */
+            return posix_error();
+        }
+
+        cwd_len = wcslen(cwd);
+        abs_path_len = cwd_len + 1 + rel_path_len;
+        if ((size_t)abs_path_len + 1 > (size_t)PY_SSIZE_T_MAX / sizeof(wchar_t)) {
+            return posix_error();
+        }
+
+        abs_path = PyMem_RawMalloc(((size_t)abs_path_len + 1) * sizeof(wchar_t));
+        if (!abs_path) {
+            return NULL;
+        }
+
+        // Join cwd & path
+        wchar_t *abs_path_dup = abs_path;
+        memcpy(abs_path_dup, cwd, cwd_len * sizeof(wchar_t));
+        abs_path_dup += cwd_len;
+
+        *abs_path_dup = SEP;
+        abs_path_dup++;
```

Comment:
I'm a bit sceptical about this line: I don't think this works when the cwd is the root.

Reply:
The add a test for that case.

Reply:
```python
>>> import os
>>> os.getcwd()
'/'
>>> os.path.abspath("")
'//'
```